test/ets_hash_ring_test.exs

```diff
@@ -154,40 +154,197 @@ defmodule ETSHashRingOperationsTest do
     assert node_name in new_nodes
   end
 
-  test "add node", %{name: name} do
-    expected_nodes = ["d" | @nodes]
-    expected_nodes_with_replicas = for node <- expected_nodes, do: {node, @default_num_replicas}
-
-    {:ok, _} = Ring.add_node(name, "d")
-    {:ok, ^expected_nodes} = Ring.get_nodes(name)
-    {:ok, ^expected_nodes_with_replicas} = Ring.get_nodes_with_replicas(name)
-    # Select a node that should now be c.
-    assert Ring.get_ring_gen(name) == {:ok, 2}
-    assert Ring.find_node(name, 1) == {:ok, "c"}
+  describe "add_node/2" do
+    test "name only", %{name: name} do
+      expected_nodes = ["d" | @nodes]
+      expected_nodes_with_replicas = for node <- expected_nodes, do: {node, @default_num_replicas}
+
+      {:ok, _} = Ring.add_node(name, "d")
+      {:ok, ^expected_nodes} = Ring.get_nodes(name)
+      {:ok, ^expected_nodes_with_replicas} = Ring.get_nodes_with_replicas(name)
+      # Select a node that should now be c.
+      assert Ring.get_ring_gen(name) == {:ok, 2}
+      assert Ring.find_node(name, 1) == {:ok, "c"}
+    end
+
+    test "name with custom replicas", %{name: name} do
+      expected_nodes = ["d" | @nodes]
+      expected_nodes_with_replicas = for node <- @nodes, do: {node, @default_num_replicas}
+      expected_nodes_with_replicas = [{"d", 200} | expected_nodes_with_replicas]
+
+      {:ok, ^expected_nodes_with_replicas} = Ring.add_node(name, "d", 200)
+      {:ok, ^expected_nodes} = Ring.get_nodes(name)
+      {:ok, ^expected_nodes_with_replicas} = Ring.get_nodes_with_replicas(name)
+      # Select a node that should now be c.
+      assert Ring.get_ring_gen(name) == {:ok, 2}
+      assert Ring.find_node(name, 1) == {:ok, "c"}
+    end
   end
 
-  test "add node with_replicas", %{name: name} do
-    expected_nodes = ["d" | @nodes]
-    expected_nodes_with_replicas = for node <- @nodes, do: {node, @default_num_replicas}
-    expected_nodes_with_replicas = [{"d", 200} | expected_nodes_with_replicas]
+  describe "add_nodes/2" do
```

Comment:
do we need to test that the `gen-2` was scheduled for gc correctly

Reply:
I can add test coverage if you'd like.